src/controller/RaceController.js

```diff
@@ -0,0 +1,55 @@
+import { Console, Random } from '@woowacourse/mission-utils';
+import RaceView from '../view/RaceView.js';
+import CarModel from '../model/CarModel.js';
+import RaceModel from '../model/RaceModel.js';
+import InputValidator from './InputValidator.js';
+import { FORWARD_THRESHOLD } from '../constants/constraints.js';
+
+class RaceController {
+  constructor() {
+    this.view = new RaceView();
+    this.race = null;
+  }
+
+  async startRace() {
+    const carNames = await this.view.getCarNames();
+    InputValidator.validateCarName(carNames);
+    const attemptCount = await this.view.getAttemptCount();
+    InputValidator.validateAttemptCount(attemptCount);
+
+    const cars = carNames.map((name) => new CarModel(name));
+    this.race = new RaceModel(cars);
+
+    this.runRace(attemptCount);
+    this.showWinners();
+  }
```

Comment:
```suggestion
  async input() {
      const carNames = await this.view.getCarNames();
      InputValidator.validateCarName(carNames);
      const attemptCount = await this.view.getAttemptCount();
      InputValidator.validateAttemptCount(attemptCount);
      
      return { carNames, attemptCount }
  }
  
  race(carNames, attemptCount) {
      const cars = carNames.map((name) => new CarModel(name));
      this.race = new RaceModel(cars);

      this.runRace(attemptCount);
  }

  async startRace() {
    const { carNames, attemptCount } = await this.input();
    
    this.race(carNames, attemptCount);

    this.showWinners();
  }
```

레이싱의 흐름이 잘 나타나있어서 이해하기 쉬웠던 거 같아요.

지금도 좋지만 한 단계 더 추상화 해보면 어떨까요?

유효성 검사, 자동차 객체 생성 과정 등을 숨겨볼 수 있을 거 같아요.

Reply:
우와 훨씬 이해도 쉽고 추상화가 잘 된 것 같아요

동현님 코드랑 리뷰로 정말 많이 배웠습니다..!
바쁘셨을텐데 정성스럽게 리뷰 달아주셔서 감사합니다!!🙇‍♀️